internal/util/util.go

```diff
@@ -77,27 +78,27 @@ func RequeueClusterOperator(c client.Client, name string) handler.MapFunc {
 	}
 }
 
-// InspectPlatformOperators iterates over all the POs on the cluster
+// InspectPlatformOperators iterates over all the POs in the list
 // and determines whether a PO is in a failing state by inspecting its status.
 // A nil return value indicates no errors were found with the POs provided.
-func InspectPlatformOperators(POList *platformv1alpha1.PlatformOperatorList) error {
-	var failingPOs []error
-	for _, po := range POList.Items {
+func InspectPlatformOperators(poList *platformv1alpha1.PlatformOperatorList) error {
+	var poErrors []error
+	for _, po := range poList.Items {
 		if err := inspectPlatformOperator(po); err != nil {
-			failingPOs = append(failingPOs, err)
+			poErrors = append(poErrors, err)
 		}
 	}
-	if len(failingPOs) > 0 {
-		return utilerror.NewAggregate(failingPOs)
-	}
-	return nil
+	return utilerror.NewAggregate(poErrors)
 }
 
 // inspectPlatformOperator is responsible for inspecting an individual platform
 // operator resource, and determining whether it's reporting any failing conditions.
 // In the case that the PO resource is expressing failing states, then an error
 // will be returned to reflect that.
 func inspectPlatformOperator(po platformv1alpha1.PlatformOperator) error {
+	if equality.Semantic.DeepEqual(po.Status, platformv1alpha1.PlatformOperatorStatus{}) {
```

Comment:
@joelanford Hi, when `equality.Semantic.DeepEqual(po.Status, platformv1alpha1.PlatformOperatorStatus{})` return True, does it mean the PO status part is empty? 
if it is correct understanding, I can not understand why `return nil` here because I think`return nil` means PO is in healthy status in method `InspectPlatformOperators `, and not report error to aggregated CO. But we do not the PO final status yet (maybe finally the PO is not installed successfully). So, here maybe `return buildPOFailureMessage(po.GetName(), "PO has no status field")`

(I just worry about that before PO installation is done (no final status yet), we report wrong information to aggregated CO. Thanks to check my concern)

Reply:
> does it mean the PO status part is empty?

Yes

> I can not understand why return nil here 

@timflannagan and I were chatting about this yesterday. In general, a PO with no status is a PO that has yet to be reconciled by the PO controller and that typically happens only as the PO is _just_ created. When the PO controller reconciles this PO, it'll attach a status and then this controller will reconcile again.

One thought I had to guard against the possibility of the PO controller never actually reconciling the PO: We could check `status is empty` and `creationTimestamp is within the last 1 minute`.

Which would result in:
1. Empty status within a minute of being created: no problem, return `nil`
2. Empty status after a minute of being create: something's wrong, return an error.

WDYT?

Reply:
@joelanford Should we add a comment that basically explains what you had written down here? I should've done that in the PR that refactored these checks.

In general, I think that option you outlined seems reasonable at a glance but introduces potential complexity.

Reply:
Thinking about this more, in the current implementation I think we'll see flapping in the aggregate ClusterOperator resource we're managing. I'm working on the POM component's implementation, which is responsible for bubbling up the underlying BundleDeployment status. In the case where the BD resource is waiting to be unpacked, we'll effectively see there's no TypeApplied status condition type present yet, and propagate the ApplyPending status condition reason in the PO resource. The aggregate CO controller would see that, and mark the aggregate CO resource with the following:

```bash
$ k get co -w
...
platform-operators-aggregated                        False       True          False      0s      encountered the failing cincinnati-operatorrwxr4 platform operator with reason "ApplyPending"
...
platform-operators-aggregated                        False       True          False      0s      encountered the failing cincinnati-operatorrwxr4 platform operator with reason "ApplyPending"
platform-operators-aggregated                        True        True          False      0s      
```

Which feels like the wrong behavior, UX, etc.

Reply:
@joelanford Thanks to check my comment.

Your idea is ok, but potential complexity (same to timflannagan).
maybe we just change `return nil` to `return buildPOFailureMessage(po.GetName(), platformtypes.ReasonApplyPending)` because we possible treat `no reconciled yet` as `ApplyPending`.

anyway I am also OK if you want to take your idea on 1minute.

Reply:
My main concern is that it seems misleading and alarmist to say a PO is failing when in fact it has just been created and is waiting for the PO controller to do something with it.

In the most recent commit, I've updated to include the "time since creationTimestamp" check. But if there's another approach that lets us avoid that complexity AND avoid saying the PO is failing, I'm totally willing to get rid of that.